napari/layers/labels/_tests/test_labels_mouse_bindings.py

```diff
@@ -39,19 +47,40 @@ def test_paint(Event):
 
     # Simulate click
     event = ReadOnlyWrapper(
-        Event(type='mouse_press', is_dragging=False, position=(0, 0))
+        Event(
+            type='mouse_press',
+            is_dragging=False,
+            position=(0, 0),
+            view_direction=(0, 1, 0),
+            dims_displayed=(0, 1),
+            dims_point=(0, 0),
+        )
     )
     mouse_press_callbacks(layer, event)
 
     # Simulate drag
     event = ReadOnlyWrapper(
-        Event(type='mouse_move', is_dragging=True, position=(19, 19))
+        Event(
+            type='mouse_move',
+            is_dragging=True,
+            position=(19, 19),
+            view_direction=(0, 1, 0),
```

Comment:
This is an odd choice of view direction for a 2D view? I'd probably go for (1, 0, 0)?

Reply:
I think we should probably change this behavior.  The view direction in world coordinates doesn't really make sense conceptually when everything is 2D since our world is 2D in that case. Perhaps it should be `None` in 2D?

Reply:
Yup! I *almost* wrote that instead! 😂

Reply:
I think we can either update the line below to be `None` if the `self.viewer.dims.ndisplay ==2` or, we can update `get_view_direction_in_scene_coordinates` to return `None` if len(dims_displayed) == 2`. Unfortunately, I won't have time to get to that before the weekend, but I can do it on Monday! Otherwise, feel free to have a go at it if you have time, @alisterburt .

https://github.com/napari/napari/blob/3bd58bde6858eb4d9f916fe32df262a4ebe1a24f/napari/_qt/qt_viewer.py#L807-L809

Reply:
Done!

Reply:
@kevinyamauchi this doesn't seem done to me? ie you are still creating some weird view_direction in 2D here that shouldn't be there?